src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/services/ExceptionRecordToCaseTransformerTest.java

```diff
@@ -0,0 +1,135 @@
+            .hasMessageContaining("Missing required fields")
+            .hasMessageContaining(OcrFieldNames.FIRST_NAME)
+            .hasMessageContaining(OcrFieldNames.LAST_NAME);
+    }
+
+    @Test
+    public void should_map_exception_record_to_a_case() {
+        // given
+        ScannedDocument erDoc1 = new ScannedDocument(
+            "type1",
+            "subtype1",
+            "url1",
+            "dcn1",
+            "filename1",
+            now(),
+            now()
+        );
+
+        ScannedDocument erDoc2 = new ScannedDocument(
+            "type2",
+            "subtype2",
+            "url2",
+            "dcn2",
+            "filename2",
+            now(),
+            now()
+        );
+
+
+        ExceptionRecord exceptionRecord = new ExceptionRecord(
+            "er-id",
+            "er-case-type",
+            "er-pobox",
+            "er-jurisdiction",
+            JourneyClassification.NEW_APPLICATION,
+            now(),
+            now(),
+            asList(
+                erDoc1,
+                erDoc2
+            ),
+            asList(
+                new OcrDataField(OcrFieldNames.FIRST_NAME, "John"),
+                new OcrDataField(OcrFieldNames.LAST_NAME, "Smith"),
+                new OcrDataField(OcrFieldNames.ADDRESS_LINE_1, "address_line_1"),
+                new OcrDataField(OcrFieldNames.ADDRESS_LINE_2, "address_line_2"),
+                new OcrDataField(OcrFieldNames.ADDRESS_LINE_3, "address_line_3"),
+                new OcrDataField(OcrFieldNames.COUNTY, "county")
+            )
+        );
+
+        // when
+        SuccessfulTransformationResponse result = service.toCase(exceptionRecord);
+        SampleCase caseData = result.caseCreationDetails.caseData;
+
+        // then
+        assertThat(result.warnings).isEmpty();
+        assertThat(result.caseCreationDetails.caseTypeId).isEqualTo(ExceptionRecordToCaseTransformer.CASE_TYPE_ID);
+        assertThat(result.caseCreationDetails.eventId).isEqualTo(ExceptionRecordToCaseTransformer.EVENT_ID);
+
+        assertThat(caseData.firstName).isEqualTo("John");
+        assertThat(caseData.lastName).isEqualTo("Smith");
+        assertThat(caseData.address.addressLine1).isEqualTo("address_line_1");
+        assertThat(caseData.address.addressLine2).isEqualTo("address_line_2");
+        assertThat(caseData.address.addressLine3).isEqualTo("address_line_3");
+        assertThat(caseData.address.county).isEqualTo("county");
+
+        assertThat(caseData.scannedDocuments).hasSize(2);
+        assertThat(caseData.scannedDocuments)
+            .as("Documents should reference source exception record")
+            .extracting(doc -> doc.value.exceptionRecordReference)
+            .containsExactly(exceptionRecord.id, exceptionRecord.id);
+
+        checkScannedDoc(caseData.scannedDocuments.get(0).value, erDoc1);
+        checkScannedDoc(caseData.scannedDocuments.get(1).value, erDoc2);
+    }
+
+    private void checkScannedDoc(
```

Comment:
we can do something like below instead of get(0) and get(1)
```assertThat(fellowshipOfTheRing).extracting("name", "age", "race.name")
                               .contains(tuple("Boromir", 37, "Man"),
                                         tuple("Sam", 38, "Hobbit"),
                                         tuple("Legolas", 1000, "Elf"));```

Reply:
I've got 8 fields to compare, so not so sure

Reply:
```
            .extracting(
                doc -> tuple(
                    doc.controlNumber,
                    doc.type,
                    doc.subtype,
                    doc.fileName,
                    doc.url,
                    doc.deliveryDate,
                    doc.scannedDate,
                    doc.exceptionRecordReference
                ))
            .containsExactly(
                tuple(
                    erDoc1.controlNumber,
                    erDoc1.type,
                    erDoc1.subtype,
                    erDoc1.fileName,
                    erDoc1.url,
                    erDoc1.deliveryDate,
                    erDoc1.scannedDate,
                    exceptionRecord.id
                ),
                tuple(
                    erDoc2.controlNumber,
                    erDoc2.type,
                    erDoc2.subtype,
                    erDoc2.fileName,
                    erDoc2.url,
                    erDoc2.deliveryDate,
                    erDoc2.scannedDate,
                    exceptionRecord.id
                )
            );
```

Reply:
I've extracted validation to a separate component with its' own unit tests